### ISSUE TITLE
discussion: remove "send email" option for public links

### ISSUE THREAD

**Issue by schiessle** [nextcloud/server#1916]
If you create a public link you can send this link directly by mail to your friend or colleague by entering the email address to a input field _if_ this option is enabled by the admin, by default it is disabled.

I want to discuss the idea of removing this functionality because of following reasons:

This feature was introduced because it was the only option to send a file directly by mail to someone outside of your Nextcloud and it was requested by some large installations. With Nextcloud 11 it will be possible to send share links by mail directly from the share dialogue (https://github.com/nextcloud/server/pull/657), this has many advantages:
- The user see to whom he send a link by mail
- The user can adjust the permissions for every recipient individually
- The user can remove links for a single recipient without affecting other recipients

On the other hand the "send public link by mail" function has many drawbacks:
-  Every recipient will receive the same link
- If you open the share dialog later you don't see to whom you already send the link by mail
- If you want to remove access to the file for one recipient you have to remove it from all, because there is only one public link

Beside that this feature is broken at least since Nextcloud 9 and nobody noticed it which is a indicator that only a few people enables this feature and even less people use it. The share dialog has already a lot (to many) options. With Nextcloud 11 we can clean this up by removing one admin setting and a input field in the share dialog without losing functionality. As a side effect it safes us a lot of time we would otherwise spend to fix the feature without breaking the new "Share by Mail" feature (which seems to be not that easy). 

Opinions: @rullzer @nickvergessen @karlitschek @LukasReschke @jancborchardt @blizzz @icewind1991 @ChristophWurst @georgehrke and everybody I forgot.

**Comment by rullzer** [comment]
I like it!
Of course https://github.com/nextcloud/server/pull/657 then needs to allow for setting all the general link sharing stuff (expiration date, password?).

But as far as I know @nickvergessen wan't expiring shares anyway. So that could work out.

**Comment by schiessle** [comment]
I don't think #657 will have expiration date or password. That's also not available for other shares creates from the share entry. If we introduces expire dates for all shares this is something different. But this would be out of scope for #657 and would first need a UI design.

**Comment by LukasReschke** [comment]
Seconding. Sounds good to me. 👍

**Comment by karlitschek** [comment]
makes sense to me. 👍  But it is important that the user understand that they can type in an email. We need some indication for this. Not sure if it exists already.

**Comment by nickvergessen** [comment]
We have a placeholder there, which can be extended easily `Share with users, groups or remote users…`

**Comment by schiessle** [comment]
> We have a placeholder there, which can be extended easily

Which is already part of my PR :smile:

**Comment by blizzz** [comment]
kill it :crossed_swords: :skull:

**Comment by MorrisJobke** [comment]
We also got following feedback: don't send that email in the name of the user (this is at least in Germany not allowed). We should use the no-reply email address ;)

**Comment by schiessle** [comment]
OK, seems like we reached a conclusion. I will take care of it once #657 is merged to avoid complicated merge conflicts.

**Comment by jancborchardt** [comment]
Thought I replied: @schiessle awesome on removing this, I always wanted to get rid of it in favor of the proper »Share by email«. Thanks a lot for implementing that! :)
